pkg/pquerier/frames.go

```diff
@@ -332,6 +347,54 @@ func (d *dataFrame) TimeSeries(i int) (utils.Series, error) {
 	}
 }
 
+// Creates Frames.columns out of tsdb columns.
+// First do all the concrete columns and then the virtual who are dependant on the concrete.
+func (d *dataFrame) finishAllColumns() error {
+
+	// Marking as Deleted all the indexes that has no data.
+	for i, hasData := range d.nonEmptyRowsIndicator {
+		if !hasData {
+			for _, col := range d.columns {
+				_ = col.Delete(i)
+			}
+			d.index.Delete(i)
+		}
+	}
+
+	var columnSize int
+	var err error
+	for _, col := range d.columns {
+		switch col.(type) {
+		case *dataColumn:
+			err = col.finish()
+		case *ConcreteColumn:
+			err = col.finish()
+			if columnSize == 0 {
+				columnSize = col.FramesColumn().Len()
+			} else if columnSize != col.FramesColumn().Len() {
+				return fmt.Errorf("columns length mismatch %v!=%v col=%v", columnSize, col.FramesColumn().Len(), col.Name())
```

Comment:
Looks like "column length"

Reply:
it's `columns`, since the length of 2 or more columns was not the same.